### DISCUSSION TITLE
Lazer song search shouldn't change the current playing song(?)

### DISCUSSION THREAD

**Discussion by integerrr** [ppy/osu#14564] [Ideas]
Putting it here to gauge whether other people are feeling the way I did when playing on lazer. The fact that everytime you touched the keyboard in song select, whether intentional or accidentally, always moves you to another song, and the accidental part is what's driving me nuts; to compare, in stable the song doesn't change when searching unless the search is narrowed down to a very small number of maps/song

**Comment by LittleEndu** [comment]
* Will probably be fixed by #14499

Glad to see I'm not the only one annoyed by this :P

**Reply by bdach** [reply]
I don't think this fixes that entirely. I believe the point is to *never* change the current selected beatmap even if the current one is filtered out.

**Reply by integerrr** [reply]
Was in the middle of checking the pr branch but yeah, exactly as @bdach described

**Reply by LittleEndu** [reply]
hmmm, right, stable doesn't eagerly select beatmaps unless only one beatmap is left unfiltered, god it even says that in the original comment... I was basing my comment on the "always moves you to another song, and the accidental part is what's driving me nuts" part, that would be fixed by my PR.

**Reply by integerrr** [reply]
Maybe should've worded my op better, the selection change on accidental search population is driving me nuts

**Comment by LittleEndu** [comment]
So would a possible solution, that keeps the eager selection lazer has, be returning to the now unfiltered beatmap if the beatmap selection has only changed because of search query? Like the scenario would be:
 * You have maps A-Z
 * B is selected
 * You search for X
 * X is selected
 * You remove the search query
 * B is re-selected
 * You search for XYZ
 * One of those is selected (current carousel selection is random)
 * You select Y
 * You remove the search query
 * Y is still selected

**Reply by bdach** [reply]
That seems like it'd require an unnecessarily complex implementation. I'd rather have something simpler. Like what stable does verbatim (eagerly selecting only if there is 1 result).

**Reply by integerrr** [reply]
I would prefer simply replicating stable behavior as it is working great already - no need to reinvent the wheel here imo

**Reply by LittleEndu** [reply]
Something like this? https://github.com/ppy/osu/compare/master...LittleEndu:turn-off-root-eager-selection

Except it probably has 1 million different test failures because all carousel tests are expecting eager selection

**Reply by bdach** [reply]
I suppose. Needs feedback from the team regarding this direction still.

**Reply by LittleEndu** [reply]
Implemented the method I laid out in OP that keeps eager selection too: https://github.com/ppy/osu/compare/master...LittleEndu:previous-selection-filtered

**Comment by peppy** [comment]
I need a fair while to think about this one. I didn't realise the behaviour implemented in lazer was different, and immediately I tend to prefer what we have as the selection is always visible. But I can understand the case for wanting the previous behaviour.

Stable handles this pretty bad because you can actually start play of a beatmap that isn't even shown on screen. I don't really like this.

**Reply by integerrr** [reply]
I've never had "start play of a beatmap that isn't even shown on screen" before, what do you mean by that?

**Reply by LittleEndu** [reply]
If you filter out only some of the maps and don't select anything yourself, you can start playing your previous selection. This is actually an issue in my quick proposed change too :/ :/

**Reply by integerrr** [reply]
but the previous selection should be visible on the left even if it's not shown in the carousel?

**Reply by integerrr** [reply]
quick idea is to always keep the current selection shown in the carousel but doesnt change selection unless the search filters down to 1 map

**Comment by ShaddyDC** [comment]
I've been running into this now and then as well, and it's really frustrating. I fatfingered a key next to enter when trying to start a map today, which changed the query and excluded my selection, and couldn't find it any more afterwards. 
I understand how it may seem bad if you are able to start playing a map that isn't shown on screen, though from a player perspective, it has never really bothered me.
An idea to allow easily going back to a map that was thus unselected, is to add it to the f2 queue or to a similar mechanism, so you can jump back to it with shift+f2. Of course, you'd still have to decide how to handle jumping to a map that is currently not visible. 
Maybe there could just be a list of recent maps somewhere? Maybe even a sorting option in song select?
